### PR TITLE
migrate e2e tests to pytest

### PR DESCRIPTION
**Pull request by peterthomassen**
- ~some tests fail, I can't figure out why~ done
- ~in test_api_rrset.py, I'll add some more parameterizations~ done

### CODE REVIEW

**Comment by peterthomassen** [comment]
> * some tests fail, I can't figure out why

Cause may be in cd37f87a93b1b87a325603a8e3c3eac79156f6af (perhaps race condition with undoing the faketime shift). On the other hand, shifts were otherwise accumulating across local test runs, and that had the smell of non-reproducibility. Any ideas?

**Comment by peterthomassen** [comment]
All done, ready for review! :tada:

File: test/e2e2/conftest.py [hunk 1]
```diff
@@ -413,12 +413,18 @@ def faketime_get():
...
+        self._faketime = faketime_get()
+        assert self._faketime[0] == '+'
+        assert self._faketime[-1] == 'd'
+        current_days = int(self._faketime[1:-1])
 
-    faketime(f'+{current_days + days:n}d')
+        faketime(f'+{current_days + self.days:n}d')
+
+    def __exit__(self, type, value, traceback):
+        faketime(self._faketime)
```
Comment (nils-wisiol) [inline test/e2e2/conftest.py:501]: Going back in time may confuse other components of desec-stack and could lead to *weird* seemingly nonsensical errors of subsequent tests. In other words, going back in time could have unforeseen consequences on subsequent tests, ruining the test independence. On the other hand, I can clearly see the advantage in resetting the clock to original time. So, no strong opinion here on either option, just pointing out a potential trap.

**Review by nils-wisiol — COMMENTED** [review]
There is some concern about going back in faketime, let's discuss before merging

**Comment by nils-wisiol** [comment]
The failure in the tests [can be traced to](https://github.com/desec-io/desec-stack/pull/563/checks?check_run_id=3103048513#step:6:1073):
```
Jul 19 10:46:15 fv-az129-926 desec/api[891]:   File "/usr/src/app/desecapi/management/commands/align-catalog-zone.py", line 24, in handle
Jul 19 10:46:15 fv-az129-926 desec/api[891]:     response = _pdns_get(NSMASTER, f'/zones/{catalog_zone_id}')
```

It appears that nsmaster could not start for some reason.

**Comment by nils-wisiol** [comment]
This seems to be caused by the PC renaming:

```
Jul 19 10:46:12 fv-az129-926 desec/nsmaster[891]: Jul 19 10:46:12 This is a standalone pdns
Jul 19 10:46:12 fv-az129-926 desec/nsmaster[891]: Jul 19 10:46:12 Created local state directory '/var/run/pdns/'
Jul 19 10:46:12 fv-az129-926 desec/nsmaster[891]: Jul 19 10:46:12 Listening on controlsocket in '/var/run/pdns/pdns.controlsocket'
Jul 19 10:46:12 fv-az129-926 desec/nsmaster[891]: Jul 19 10:46:12 Fatal error: Trying to set unknown parameter 'secondary'
```

**Comment by peterthomassen** [comment]
That's because this PR is not yet rebased on the #566. It used to have this combined, but did the split once you were asking for a minimal PR :) (the result is #566, and this one being temporarily broken)

File: test/e2e2/conftest.py (hunk 1 above)
Comment (peterthomassen) [reply test/e2e2/conftest.py:501]: Fair enough. I'd say, let's see if it causes problems :)

Comment (nils-wisiol) [reply test/e2e2/conftest.py:501]: Alright let's add a usage warning (code comment) to the context manager

**Review by nils-wisiol — APPROVED** [review]

File: test/e2e2/conftest.py (hunk 1 above)
Comment (peterthomassen) [reply test/e2e2/conftest.py:504]: done

**Pull request merged**